dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/extensions/SecurityContextImpl.java

```diff
@@ -118,8 +122,41 @@ public Principal getCallerPrincipal() {
      * @see javax.security.enterprise.SecurityContext#hasAccessToWebResource(java.lang.String, java.lang.String[])
      */
     @Override
-    public boolean hasAccessToWebResource(String arg0, String... arg1) {
-        // TODO Auto-generated method stub
+    public boolean hasAccessToWebResource(String resource, String... methods) {
+
+        String appName = getApplicationName();
+        SecurityMetadata securityMetadata = getSecurityMetadata();
+        SecurityConstraintCollection collection = securityMetadata.getSecurityConstraintCollection();
+
+        if (null != collection) {
+            AuthorizationService authService = SecurityContextHelper.getAuthorizationService();
+            Subject callerSubject = getCallerSubject();
+
+            for (String method : methods) {
+                MatchResponse matchResponse = collection.getMatchResponse(resource, method);
+
+                if (matchResponse.equals(MatchResponse.NO_MATCH_RESPONSE)) {
+                    // not match responses,  continue to the next method
```

Comment:
If there is no security constraint for that method, then the user has access to it because it's unprotected and we need to return true from here.